data_structures/src/chain.rs

```diff
@@ -923,6 +923,34 @@ impl TransactionsPool {
         self.re_transactions.clear();
     }
 
+    /// Returns `true` if the pool contains a value transfer
+    /// transaction for the specified hash.
+    pub fn contains(&self, transaction: &Transaction) -> bool {
+        let tx_hash = transaction.hash();
+
+        match transaction {
+            Transaction::ValueTransfer(_vt) => self.vt_contains(&tx_hash),
+            Transaction::DataRequest(_drt) => self.dr_contains(&tx_hash),
+            Transaction::Commit(ct) => {
+                let dr_pointer = ct.body.dr_pointer;
+
+                if let Some(pkh) = &ct.signatures.get(0).map(|x| x.public_key.pkh()) {
```

Comment:
Maybe it would be better to use:

    let pkh = ct.body.proof.proof.pkh();

Reply:
it should be the same